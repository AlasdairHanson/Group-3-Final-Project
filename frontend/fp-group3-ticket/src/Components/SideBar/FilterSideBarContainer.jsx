import React from "react";
import { Accordion, Card } from "react-bootstrap";
import SideFilterTopic from "./SideFilterTopic";
import SideFilterPriority from "./SideFilterPriority";
import SideFilterStatus from "./SideFilterStatus";
import SideFilterCohort from "./SideFilterCohort";
import SideHeader from "./SideHeader";
const FilterSideBarContainer = ({ dataSortSetup }) => {
  return (
    <div className="sidebarContainer mb-3">
      <Accordion>
        <Card className="sideBarContainer">
          <Accordion.Toggle
            as={Card.Header}
            eventKey="0"
            className="p-0 sideBarContainer"
          >
            <SideHeader title="Filter" />
          </Accordion.Toggle>
          <Accordion.Collapse eventKey="0">
            <Card.Body className="p-0">
              <SideFilterPriority dataSortSetup={dataSortSetup} />
              <SideFilterStatus dataSortSetup={dataSortSetup} />
              <SideFilterTopic dataSortSetup={dataSortSetup} />
              <SideFilterCohort dataSortSetup={dataSortSetup} />
            </Card.Body>
          </Accordion.Collapse>
        </Card>
      </Accordion>
    </div>
  );
};
export default FilterSideBarContainer;
