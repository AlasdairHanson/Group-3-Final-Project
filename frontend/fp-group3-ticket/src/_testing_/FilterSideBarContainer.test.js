import { render, screen } from '@testing-library/react';
import FilterSideBarContainer from '../Components/SideBar/FilterSideBarContainer';
import '@testing-library/jest-dom';
import {create} from 'react-test-renderer';

jest.mock(`../Components/SideBar/SideFilterPriority`, () => {
  return function FilterPriorityMock(props){
    return(<p>Mocked FilterPriority</p>)
  }
});

jest.mock(`../Components/SideBar/SideFilterCohort`, () => {
    return function FilterCohortMock(props){
      return(<p>Mocked FilterCohort</p>)
    }
  });

jest.mock(`../Components/SideBar/SideFilterStatus`, () => {
    return function FilterStatusMock(props){
      return(<p>Mocked FilterStatus</p>)
    }
  });

jest.mock(`../Components/SideBar/SideFilterTopic`, () => {
    return function FilterTopicMock(props){
      return(<p>Mocked FilterTopic</p>)
    }
  });

jest.mock(`../Components/SideBar/SideHeader`, () => {
    return function FilterTopicMock(props){
      return(<p>Mocked FilterTopic</p>)
    }
  });

describe(`Snapshot test of FilterSideBarContainer`, () => {

  test(` FilterSideBarContainer should match the snapshot every time it is rendered`, () => {
    const instance = create(<FilterSideBarContainer/>).toJSON(); 
    expect(instance).toMatchSnapshot();
  });
});