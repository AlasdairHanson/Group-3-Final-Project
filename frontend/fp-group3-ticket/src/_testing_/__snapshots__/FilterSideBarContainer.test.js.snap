// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of FilterSideBarContainer  FilterSideBarContainer should match the snapshot every time it is rendered 1`] = `
<div
  className="sidebarContainer mb-3"
>
  <div
    className="accordion"
  >
    <div
      className="sideBarContainer card"
    >
      <div
        className="p-0 sideBarContainer card-header"
        onClick={[Function]}
      >
        <p>
          Mocked FilterTopic
        </p>
      </div>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="p-0 card-body"
        >
          <p>
            Mocked FilterPriority
          </p>
          <p>
            Mocked FilterStatus
          </p>
          <p>
            Mocked FilterTopic
          </p>
          <p>
            Mocked FilterCohort
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
